Automattic/wp-calypso
# chore: Extract babel config to a separate package `@automattic/calypso-babel-config`

#### Changes proposed in this Pull Request

Extracts Babel config to its own package, `@automattic/calypso-babel-config`.

This package provides both the main config used in Calypso (i.e. the content of `babel.config.js`). This allows us to have a single place to centralize babel config and dependencies, instead of having them scattered across multiple projects.

This new package also provides the same presets that were available in `@automattic/calypso-build/presets/*`. This should allow consumers (cc @Automattic/jetpack-crew ) to depend on `@automattic/calypso-babel-config` when needed, instead of pulling in the whole `@automattic/calypso-build` and its peer dependencies.

Depends on #57222

#### Testing instructions

Note: the instructions assume the PR has been rebased with `trunk`. If that's not the case, use `d0765fe5f9` as the commit to compare against (the last common ancestor at the time of writing)

* Checkout this branch, build Calypso, and verify the build output has not changed compared with `trunk`.
    * Build it with `NODE_ENV=production CALYPSO_ENV=production BUILD_TRANSLATION_CHUNKS=true yarn build`
    * Compare them with `diff -qr wp-calypso/public/evergreen wp-calypso-trunk/public/evergreen`
* Checkout this branch, build ETK, and verify the build output has not changed compared with `trunk`.
    * Build it with `cd apps/editing-toolkit && yarn build`
    * Compare them with `diff -qr wp-calypso/apps/editing-toolkit/editing-toolkit-plugin wp-calypso-trunk/apps/editing-toolkit/editing-toolkit-plugin`
* Checkout this branch, build Notifications, and verify the build output has not changed compared with `trunk`.
    * Build it with `cd apps/notifications && yarn build`
    * Compare them with `diff -qr wp-calypso/apps/notifications/dist wp-calypso2/apps/notifications/dist`. There will be some minor differences related to commit hash.
* Checkout this branch, build o2-blocks, and verify the build output has not changed compared with `trunk`.
    * Build it with `cd apps/o2-blocks && yarn build`
    * Compare them with `diff -qr wp-calypso/apps/o2-blocks/dist wp-calypso2/apps/o2-blocks/dist`
* Checkout this branch, build wpcom-block-editor, and verify the build output has not changed compared with `trunk`.
    * Build it with `cd apps/wpcom-block-editor && yarn build`
    * Compare them with `diff -qr wp-calypso/apps/wpcom-block-editor/dist wp-calypso2/apps/wpcom-block-editor/dist`
